__tests__/rpcProvider.test.ts

```diff
@@ -50,6 +50,17 @@ describeIfRpc('RPCProvider', () => {
     await createBlockForDevnet();
   });
 
+  test('instantiate from rpcProvider', () => {
+    // instantiate rpc provider template
+    const newInsRPCProvider = new RpcProvider();
+    // like* change channel and parser
+    newInsRPCProvider.channel = rpcProvider.channel;
+    newInsRPCProvider.responseParser = rpcProvider.responseParser;
+    // instantiate from modified instance
+    const FinalInsRPCProvider = new RpcProvider(newInsRPCProvider);
+    expect(FinalInsRPCProvider).toBeInstanceOf(RpcProvider);
```

Comment:
I would suggest changing the test to something like this so both branches of the parser conditional are covered:

```suggestion
    const newInsRPCProvider = new RpcProvider();
    
    let FinalInsRPCProvider = new RpcProvider(newInsRPCProvider);
    expect(FinalInsRPCProvider.channel).toBe(newInsRPCProvider.channel);
    expect(FinalInsRPCProvider.responseParser).toBe(newInsRPCProvider.responseParser);
    
    delete (newInsRPCProvider as any).responseParser;
    FinalInsRPCProvider = new RpcProvider(newInsRPCProvider);
    expect(FinalInsRPCProvider.channel).toBe(newInsRPCProvider.channel);
    expect(FinalInsRPCProvider.responseParser).toBeInstanceOf(RPCResponseParser);
```